Realm/RLMBSON.mm

```diff
@@ -370,7 +370,7 @@ Bson RLMConvertRLMBSONToBson(id<RLMBSON> b) {
         case realm::bson::Bson::Type::Double:
             return @(static_cast<double>(b));
         case realm::bson::Bson::Type::String:
-            return RLMStringDataToNSString(static_cast<std::string>(b).c_str());
+            return [NSString stringWithUTF8String:static_cast<std::string>(b).c_str()];
```

Comment:
This ensures we copy the bytes. We also didn't need `RLMStringDataToNSString` as the value comes in as a `std::string` so we performed an extra conversion too.

Reply:
RLMStringDataToNSString also copies the input data. Since this doesn't need to handle embedded nulls, this should be just `@(static_cast<std::string>(b).c_str())`.